### DISCUSSION TITLE
Declaring an id with cuid() doesn't satisfy unique requirement of connect API

### DISCUSSION THREAD

**Discussion by sp88011** [prisma/prisma#18278] [Q&A] [answered]
I'm new to Prisma so it's possible this isn't a bug but it strikes me as one.

The `connect` attribute inside a `create` function expects a unique identifier for the relation that you want to connect the newly created element with. Makes sense.

I was surprised to find that declaring a field as `@id @default (cuid())` in my schema does not mean that Prisma treats that field as unique. 

Example MySQL:

```prisma
//schema.prisma
model FirstName {
  id                    String                  @id @default(cuid())
  createdAt      DateTime            @default(now())
  name             String
  LastName     LastName[]
  
  @@unique([id]) // this is required to make the code inside page.ts work.  But it shouldn't be needed since cuid() ensures the field is unique
}

model LastName {
  id                    String                  @id @default(cuid())
  createdAt      DateTime             @default(now())
  name             String
  FirstName     FirstName[]
 }

```

```ts
//page.ts
await prisma.firstname.create({
data:  { name: "John", 
LastName: {
 connect: [{id:1}, {id:100}, {id:222}]
 }
}
})

```

**Comment by nurul3101** [comment]
Hello @sp88011 👋 

Can you share the error message that you got when trying to connect lastname records?
Also, to confirm, the lastname records with id 1,100,222 were already created in database, right?

**Reply by nurul3101** [reply, marked as answer]
@sp88011 

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
